### PR TITLE
Generic cpu memory alert rules

### PR DESCRIPTION
**Pull request by rakesh-garimella**
**Description**

Alert rules when CPU and Memory are more than 90% used

One needs to add label `alertmem: yes` for subscribing to alerts on memory for a pod and `alertcpu:yes` for alerts on CPU

* Move all the alerts into a single file `kyma-alerts` since only one `tpl` file is allowed. So add all the kyma related rules into `kyma-alerts`

**Related issue(s)**
Fixes: #1817

### CODE REVIEW

File: resources/monitoring/charts/alert-rules/templates/kyma-rules.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+- name: cpu-90-percent-usage-rule
+  rules:
+  - alert: CPU90PercentUsage
+    expr:   sum (rate(container_cpu_usage_seconds_total{namespace="kyma-system"}[2m])) by (pod_name) * on(pod_name) group_left(app) label_replace(max(kube_pod_labels{label_alertcpu=~"yes*"}) by (pod), "pod_name", "$1", "pod", "(.*)") * 100 > 90
+    for: 10m
+    labels:
+      severity: critical
+    annotations:
+      description:  "Pod {{`{{$labels.pod}}`}} is having {{`{{$value}}`}} % usage"
+      summary: "Pod {{`{{$labels.pod}}`}} is having {{`{{$value}}`}} % usage"
```
Comment (sayanh) [inline resources/monitoring/charts/alert-rules/templates/kyma-rules.yaml]: screen shot 2019-02-06 at 16 48 04
Looks like pod name and namespace are not appearing in `CPU90Percentage` and `MEM90PercentUsage` alerts.

File: resources/monitoring/charts/alert-rules/README.md [hunk 2]
```diff
@@ -46,10 +46,28 @@ data:
...
+
+* Monitoring Persistent Volume Claims (PVC)
+
+  The Alertmanager triggers the rule when PVC in any of the system Namespaces such as `kyma-system`, `kyma-integration`, `heptio-ark`, `istio-system`, `kube-public` or `kube-system` exceeds  90%. In such a case, increase the capacity of PVCs.
+
+* Monitoring CPU Usage
+
+  The Alertmanager trigger the rule when CPU usage is above 90% in the pods in the namespace `kyma-system`. For the alert rule to fire for a pod its necessary that label `alertcpu: "yes"` is passed to the pods.
+
+* Monitoring Memrory usage
```
Comment (sayanh) [inline resources/monitoring/charts/alert-rules/README.md]: typo: `Memory`

File: resources/monitoring/charts/alert-rules/README.md [hunk 3]
```diff
@@ -46,10 +46,28 @@ data:
...
+
+   The Alertmanager sends out alerts when one of the pods is not running in any of the system Namespaces such as `kyma-system`, `kyma-integration`, `istio-system`, `kube-public` or `kube-system`
+
+* Monitoring Persistent Volume Claims (PVC)
+
+  The Alertmanager triggers the rule when PVC in any of the system Namespaces such as `kyma-system`, `kyma-integration`, `heptio-ark`, `istio-system`, `kube-public` or `kube-system` exceeds  90%. In such a case, increase the capacity of PVCs.
+
+* Monitoring CPU Usage
+
+  The Alertmanager trigger the rule when CPU usage is above 90% in the pods in the namespace `kyma-system`. For the alert rule to fire for a pod its necessary that label `alertcpu: "yes"` is passed to the pods.
```
Comment (sayanh) [inline resources/monitoring/charts/alert-rules/README.md]: typo: `it's necessary`

File: resources/monitoring/charts/alert-rules/templates/kyma-rules.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    for: 60s
+    labels:
+      severity: critical
+    annotations:
+      description: "{{`{{ $labels.namespace }}`}}/{{`{{ $labels.pod }}`}} is not running"
+      summary: "{{`{{ $labels.pod }}`}} is not running"
+- name: cpu-90-percent-usage-rule
+  rules:
+  - alert: CPU90PercentUsage
+    expr:   sum (rate(container_cpu_usage_seconds_total{namespace="kyma-system"}[2m])) by (pod_name) * on(pod_name) group_left(app) label_replace(max(kube_pod_labels{label_alertcpu=~"yes*"}) by (pod), "pod_name", "$1", "pod", "(.*)") * 100 > 90
```
Comment (sayanh) [inline resources/monitoring/charts/alert-rules/templates/kyma-rules.yaml]: Why not check for: `label_alertcpu="yes"`? IMO it should be a strict match.

File: resources/monitoring/charts/alert-rules/templates/kyma-rules.yaml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    for: 10m
+    labels:
+      severity: critical
+    annotations:
+      description:  "Pod {{`{{$labels.pod}}`}} is having {{`{{$value}}`}} % usage"
+      summary: "Pod {{`{{$labels.pod}}`}} is having {{`{{$value}}`}} % usage"
+- name: mem-90-percent-usage-rule
+  rules:
+  - alert: MEM90PercentUsage
+    expr:   sum(container_memory_working_set_bytes{namespace="kyma-system"}) by (pod_name) / sum(label_join(kube_pod_container_resource_limits_memory_bytes, "pod_name", "", "pod")) by (pod_name)  * on(pod_name) group_left(app)   label_replace(max(kube_pod_labels{label_alertmem=~"yes*"}) by (pod), "pod_name", "$1", "pod", "(.*)") * 100 > 90
```
Comment (sayanh) [inline resources/monitoring/charts/alert-rules/templates/kyma-rules.yaml]: Why not check for: `label_alertcpu="yes"`? IMO it should be a strict match.

File: resources/monitoring/charts/alert-rules/README.md (hunk 3 above)
Comment (bszwarc) [inline resources/monitoring/charts/alert-rules/README.md]:
```suggestion
  The Alertmanager triggers the rule when CPU usage exceeds 90%. for Pods in the `kyma-system` Namespace. For the alert rule to activate, make sure to pass the `alertcpu: "yes"` label to Pods.
```
Question: is passed as a parameter? or the Pods should be labelled with it?

File: resources/monitoring/charts/alert-rules/README.md [hunk 6]
```diff
@@ -46,10 +46,28 @@ data:
...
 {{ else }}
   alert.rules: |-
-    {{- include "unhealthy-pods-rules.yaml.tpl" . | indent 4}}
+    {{- include "kyma-rules.yaml.tpl" . | indent 4}}
 {{ end }}
 ```
-Under the **data. alert.rules** parameter, there is a configuration of the [unhealthy-pods-rules.yaml](templates/unhealthy-pods-rules.yaml) file, which creates a rule for alerting when a Pod is not running.
+Under the **data. alert.rules** parameter, there is a configuration of the [kyma-rules.yaml](templates/kyma-rules.yaml) file, which creates a rule for following cases:
+
+*  Alerting when a Pod is not running
```
Comment (bszwarc) [inline resources/monitoring/charts/alert-rules/README.md]:
```suggestion
*  Alert when a Pod is not running
```

File: resources/monitoring/charts/alert-rules/README.md [hunk 7]
```diff
@@ -46,10 +46,28 @@ data:
...
+
+   The Alertmanager sends out alerts when one of the pods is not running in any of the system Namespaces such as `kyma-system`, `kyma-integration`, `istio-system`, `kube-public` or `kube-system`
+
+* Monitoring Persistent Volume Claims (PVC)
```
Comment (bszwarc) [inline resources/monitoring/charts/alert-rules/README.md]:
```suggestion
* Monitor Persistent Volume Claims (PVC)
```

File: resources/monitoring/charts/alert-rules/README.md [hunk 8]
```diff
@@ -46,10 +46,28 @@ data:
...
+
+  The Alertmanager triggers the rule when PVC in any of the system Namespaces such as `kyma-system`, `kyma-integration`, `heptio-ark`, `istio-system`, `kube-public` or `kube-system` exceeds  90%. In such a case, increase the capacity of PVCs.
+
+* Monitoring CPU Usage
```
Comment (bszwarc) [inline resources/monitoring/charts/alert-rules/README.md]:
```suggestion
* Monitor CPU Usage
```

File: resources/monitoring/charts/alert-rules/README.md [hunk 9]
```diff
@@ -46,10 +46,28 @@ data:
...
+
+  The Alertmanager trigger the rule when CPU usage is above 90% in the pods in the namespace `kyma-system`. For the alert rule to fire for a pod its necessary that label `alertcpu: "yes"` is passed to the pods.
+
+* Monitoring Memory usage
```
Comment (bszwarc) [inline resources/monitoring/charts/alert-rules/README.md]:
```suggestion
* Monitor Memory usage
```

File: resources/monitoring/charts/alert-rules/README.md [hunk 10]
```diff
@@ -46,10 +46,28 @@ data:
...
+
+  The Alertmanager trigger the rule when Memory usage is above 90% in the pods in the namespace `kyma-system`. For the alert rule to fire for a pod its necessary that label `alertmem: "yes"` is passed to the pods.
```
Comment (bszwarc) [inline resources/monitoring/charts/alert-rules/README.md]:
```suggestion
  The Alertmanager triggers the rule when Memory usage exceeds 90% for Pods in the `kyma-system` Namespace. For the alert to activate,  make sure you passed the `alertmem: "yes"` label to Pods.

File: resources/monitoring/charts/alert-rules/README.md [hunk 11]
```diff
@@ -46,10 +46,28 @@ data:
...
+
+An example for alerting when a pod is not running is shown below
```
Comment (bszwarc) [inline resources/monitoring/charts/alert-rules/README.md]:
```suggestion
This is an example for alerting when a Pod is not running:
```

File: resources/monitoring/charts/alert-rules/README.md [hunk 12]
```diff
@@ -46,10 +46,28 @@ data:
...
-    {{- include "unhealthy-pods-rules.yaml.tpl" . | indent 4}}
+    {{- include "kyma-rules.yaml.tpl" . | indent 4}}
 {{ end }}
 ```
-Under the **data. alert.rules** parameter, there is a configuration of the [unhealthy-pods-rules.yaml](templates/unhealthy-pods-rules.yaml) file, which creates a rule for alerting when a Pod is not running.
+The **data. alert.rules** parameter includes the configuration in [kyma-rules.yaml](templates/kyma-rules.yaml) file, which creates rules in the following cases:
+
+*  Alert when a Pod is not running
+
+  The Alertmanager sends out alerts when one of the Pods is not running in `kyma-system`, `kyma-integration`, `istio-system`, `kube-public` or `kube-system` Namespaces.
```
Comment (bszwarc) [inline resources/monitoring/charts/alert-rules/README.md]:
```suggestion
  The Alertmanager sends out alerts when one of the Pods is not running in `kyma-system`, `kyma-integration`, `istio-system`, `kube-public`, or `kube-system` Namespaces.
```

File: resources/monitoring/charts/alert-rules/README.md [hunk 13]
```diff
@@ -46,10 +46,28 @@ data:
...
+
+* Monitor Persistent Volume Claims (PVC)
+
+  The Alertmanager triggers the rule when PVC exceeds  90%  for the following system Namespaces: `kyma-system`, `kyma-integration`, `heptio-ark`, `istio-system`, `kube-public` or `kube-system`. In such a case, increase the capacity of PVC.
```
Comment (bszwarc) [inline resources/monitoring/charts/alert-rules/README.md]:
```suggestion
  The Alertmanager triggers the rule when PVC exceeds  90%  for the following system Namespaces: `kyma-system`, `kyma-integration`, `heptio-ark`, `istio-system`, `kube-public`, or `kube-system`. In such a case, increase the capacity of PVC.
```

File: resources/monitoring/charts/alert-rules/README.md [hunk 14]
```diff
@@ -46,10 +46,28 @@ data:
...
+
+* Monitor CPU Usage
+
+  The Alertmanager triggers the rule when CPU usage exceeds 90%. for Pods in the `kyma-system` Namespace. For the alert rule to activate, make sure to pass the `alertcpu: "yes"` label to Pods.
```
Comment (bszwarc) [inline resources/monitoring/charts/alert-rules/README.md]:
```suggestion
  The Alertmanager triggers the rule when CPU usage exceeds 90% for Pods in the `kyma-system` Namespace. For the alert rule to activate, make sure to pass the `alertcpu: "yes"` label to Pods.
```

File: resources/monitoring/charts/alert-rules/README.md [hunk 15]
```diff
@@ -46,10 +46,28 @@ data:
...
+
+* Monitor Memory usage
+
+  The Alertmanager triggers the rule when Memory usage is above 90% in the pods in the namespace `kyma-system`. For the alert rule to fire for a pod its necessary that label `alertmem: "yes"` is passed to the pods.
```
Comment (bszwarc) [inline resources/monitoring/charts/alert-rules/README.md]:
```suggestion
  The Alertmanager triggers the rule when Memory usage exceeds 90% for the Pods in the `kyma-system` Namespace. For the alert rule to activate, make sure to pass the `alertmem: "yes"` to Pods.

File: resources/monitoring/charts/alert-rules/README.md [hunk 16]
```diff
@@ -46,10 +46,28 @@ data:
 {{- end }}
 {{ else }}
   alert.rules: |-
-    {{- include "unhealthy-pods-rules.yaml.tpl" . | indent 4}}
+    {{- include "kyma-rules.yaml.tpl" . | indent 4}}
 {{ end }}
 ```
-Under the **data. alert.rules** parameter, there is a configuration of the [unhealthy-pods-rules.yaml](templates/unhealthy-pods-rules.yaml) file, which creates a rule for alerting when a Pod is not running.
+The **data. alert.rules** parameter includes the configuration in [kyma-rules.yaml](templates/kyma-rules.yaml) file, which creates rules in the following cases:
```
Comment (bszwarc) [inline resources/monitoring/charts/alert-rules/README.md]:
```suggestion
The **data. alert.rules** parameter includes the configuration in the [kyma-rules.yaml](templates/kyma-rules.yaml) file, which creates rules in the following cases:
```

File: resources/monitoring/charts/alert-rules/README.md [hunk 17]
```diff
@@ -46,10 +46,28 @@ data:
...
+
+    The Alertmanager triggers the rule when PVC exceeds  90%  for the following system Namespaces: `kyma-system`, `kyma-integration`, `heptio-ark`, `istio-system`, `kube-public`, or `kube-system`. In such a case, increase the capacity of PVC.
+
+* Monitor CPU Usage
+
+    The Alertmanager triggers the rule when CPU usage exceeds 90% for Pods in the `kyma-system` Namespace. For the alert rule to activate, make sure to pass the `alertcpu: "yes"` label to Pods.
+
+* Monitor Memory usage
+
+    The Alertmanager triggers the rule when Memory usage exceeds 90% for the Pods in the `kyma-system` Namespace. For the alert rule to activate, make sure to pass the `alertmem: "yes"` to Pods.
```
Comment (bszwarc) [inline resources/monitoring/charts/alert-rules/README.md]:
```suggestion
    The Alertmanager triggers the rule when Memory usage exceeds 90% for Pods in the `kyma-system` Namespace. For the alert rule to activate, make sure to pass the `alertmem: "yes"` to Pods.
```

File: resources/monitoring/charts/alert-rules/README.md (hunk 12 above)
Comment (rakesh-garimella) [reply resources/monitoring/charts/alert-rules/README.md]: done

File: resources/monitoring/charts/alert-rules/README.md (hunk 13 above)
Comment (rakesh-garimella) [reply resources/monitoring/charts/alert-rules/README.md]: done

File: resources/monitoring/charts/alert-rules/README.md (hunk 15 above)
Comment (rakesh-garimella) [reply resources/monitoring/charts/alert-rules/README.md]: done

File: resources/monitoring/charts/alert-rules/README.md (hunk 16 above)
Comment (rakesh-garimella) [reply resources/monitoring/charts/alert-rules/README.md]: done

File: resources/monitoring/charts/alert-rules/README.md (hunk 14 above)
Comment (rakesh-garimella) [reply resources/monitoring/charts/alert-rules/README.md]: done

**Pull request merged**